Lib/zipfile.py

```diff
@@ -484,6 +484,8 @@ def from_file(cls, filename, arcname=None):
         isdir = stat.S_ISDIR(st.st_mode)
         mtime = time.localtime(st.st_mtime)
         date_time = mtime[0:6]
+        if not strict_timestamps and date_time[0] < 1980:
+            date_time = (1980, 1, 1, 0, 0, 0)
```

Comment:
It seems like the largest year is 2107. Years after 2017 should also be clamped to 2017-12-31, no?

Reply:
fixed